js-apps/meep-frontend/src/js/containers/idc-map.js

```diff
@@ -0,0 +1,425 @@
+
+      // Add to map overlay
+      m.addTo(this.ueOverlay);
+      if (p) {
+        p.addTo(this.uePathOverlay);
+      }
+      // console.log('UE ' + id + ' added @ ' + latlng.toString());
+    } else {
+      // Update UE position & path
+      existingMarker.setLatLng(latlng);
+
+      // Update path
+      if (existingMarker.options.path) {
+        existingMarker.options.path.removeFrom(this.uePathOverlay);
+      }
+      if (p) {
+        existingMarker.options.path = p;
+        p.addTo(this.uePathOverlay);
+      }
+    }
+  }
+
+  setPoaMarker(poa) {
+    var latlng = L.latLng(L.GeoJSON.coordsToLatLng(poa.location.coordinates));
+
+    // Find existing POA marker
+    var existingMarker;
+    this.poaOverlay.eachLayer((marker) => {
+      if (marker.options.myId === poa.assetName){
+        existingMarker = marker;
+        return;
+      }
+    });
+
+    if (existingMarker === undefined) {
+      // Create new POA marker & circle
+      var c = L.circle(latlng, {
+        myId: poa.assetName,
+        radius: poa.radius,
+        opacity: '0.5'
+      });
+      var m = L.marker(latlng, {
+        myId: poa.assetName,
+        myCircle: c,
+        opacity: '0.5',
+        draggable: false
+      });
+      m.bindTooltip(poa.assetName).openTooltip();
+
+      // Click handler
+      var _this = this;
+      m.on('click', function(e) {_this.clickPoaMarker(this,e);});
+
+      // Add to map overlay
+      m.addTo(this.poaOverlay);
+      c.addTo(this.poaRangeOverlay);
+      // console.log('PoA ' + id + ' added @ ' + latlng.toString());
+    } else {
+      // Update POA position
+      existingMarker.setLatLng(latlng);
+      existingMarker.options.myCircle.setLatLng(latlng);
+    }
+  }
+
+  setComputeMarker(compute) {
+    var latlng = L.latLng(L.GeoJSON.coordsToLatLng(compute.location.coordinates));
+
+    // Find existing COMPUTE marker
+    var existingMarker;
+    this.computeOverlay.eachLayer((marker) => {
+      if (marker.options.myId === compute.assetName){
+        existingMarker = marker;
+        return;
+      }
+    });
+
+    if (existingMarker === undefined) {
+      // Creating new COMPUTE marker
+      var m = L.marker(latlng, {
+        myId: compute.assetName,
+        opacity: '0.5',
+        draggable: false
+      });
+      m.bindTooltip(compute.assetName).openTooltip();
+
+      // Click handler
+      var _this = this;
+      m.on('click', function() {_this.clickComputeMarker(this);});
+
+      // Add to map overlay
+      m.addTo(this.computeOverlay);
+      // console.log('Compute ' + id + ' added @ ' + latlng.toString());
+    } else {
+      // Update COMPUTE position
+      existingMarker.setLatLng(latlng);
+    }
+  }
+
+  // UE Marker Event Handler
+  clickUeMarker(marker) {
+    var latlng = marker.getLatLng();
+    var msg = '<b>UE: ' + marker.options.myId + '</b><br>';
+    msg += 'eopMode: ' + marker.options.eopMode + '<br>';
+    msg += 'velocity: ' + marker.options.velocity + ' m/s<br>';
```

Comment:
Instead of `eopMode` use `path-mode` LOOP/RETURN
Instead of `UE` use `id`

Reply:
Modified as suggested.